solarized.el

```diff
@@ -116,6 +116,11 @@ Related discussion: https://github.com/bbatsov/solarized-emacs/issues/158"
   :type 'boolean
   :group 'solarized)
 
+(defcustom solarized-termcolors nil
```

Comment:
Shouldn't we detect the capabilities of the terminal?

Reply:
AFAIK, the main issue is that a 256 color capable term emulator will be recognized as such, and those 256 colors do not contain the exact solarized palette, although the terminal might be configured with the official 16 colors palette as base colors. I may be wrong though.

Reply:
Yes, detection should not be trusted as the single source of truth.

Reply:
I see.

Reply:
Is there any widespread terminal that correctly displays the solarized color codes? If not what about `(not (display-graphic-p)` as a heuristic? Either the user has set the solarized palette in their terminal or he did not. 
- If he has set the solarized palette it works fine if we use the heuristic. If we don't they look bad.
- If he has not set the solarized palette the colors look bad anyway, so it does not make a difference if we select the "fake" palette using the heuristic (assuming nobody wants to use the solarized theme in a 256 terminal where the colors are only displayed approximately).